worker/jobTypes/githubPushJob.js

```diff
@@ -1,68 +1,197 @@
-
+/** this is the github-triggered processor for builds
+ *  it expects a worker.sh in the root of the repository
+ *  an example job definition lives in jobDef.json
+ */
 
 const fs = require('fs-extra');
 const workerUtils = require('../utils/utils');
+const simpleGit = require('simple-git/promise');
+const validator = require('validator');
 
-const gatsbyBuildTimeout = 60 * 45;
+const buildTimeout = 60 * 450;
 const uploadToS3Timeout = 20;
 
 process.env.STITCH_ID = 'ref_data-bnbxq';
 process.env.NAMESPACE = 'snooty/documents';
 
+const invalidJobDef = new Error('job not valid');
 
 module.exports = {
 
-    async gatsbyBuild(currentJob) {
-        workerUtils.logInMongo(currentJob, `${'    (GATSBY)'.padEnd(15)}Running Gatsby Build`);
+    //anything that is passed to an exec must be validated or sanitized
+    //we use the term sanitize here lightly -- in this instance this validates
+    sanitizeString(string) {
+        return validator.isAscii(string) &&
+        validator.matches(string, /^((\w)+[-]?(\w)+)*$/);
+    },
+
+    async sanitizeGithubPush(currentJob) {
```

Comment:
Yes the worker.js has the await but since we are effectively creating an interface in our worker.js, it's run as "safe" (which maps to the function in a mapper utility). I know, it's very java to do things this way!